modules/aws/agent-gw/README.md

```diff
@@ -0,0 +1,91 @@
+# DSF DAM Agent Gateway
+[![GitHub tag](https://img.shields.io/github/v/tag/imperva/dsfkit.svg)](https://github.com/imperva/dsfkit/tags)
+
+This Terraform module provisions a DSF DAM Agent Gateway on AWS as an EC2 instance.
+
+The Database Monitoring Gateway performs application and database monitoring, providing full visibility into how that data is actually used in the enterprise, regardless of whether it is accessed directly or indirectly via applications.
+
+## Requirements
+* Terraform v1.3.1.
+* An AWS account.
+* Network access to Management server (MX) on port 8083 (API and WebConsole)
+
+**NOTE:** In case you are not yet an Imperva customer, [please contact our team](https://www.imperva.com/contact-us/).
+
+## Resources Provisioned
+This Terraform module provisions several resources on AWS to create the DSF DAM Agent Gateway. These resources include:
+* An EC2 instance for running the software.
+* An EBS volume for storage.
+* A security groups to allow the required network access to and from the DSF instance.
+* An IAM role with relevant policies.
+* An AWS KMS.
+* An AWS Elastic Network Interface (ENI).
+
+The EC2 instance and EBS volume provide the computing and storage resources needed to run the DSF software. The security group controls the inbound and outbound traffic to the instance, while the IAM role grants the necessary permissions to access AWS resources. The KMS is used for encrypting sensitive data.
+
+## Inputs
+
+The following input variables are **required**:
+
+* `subnet_id`: The ID of the subnet in which to launch the DSF instance
+* `key_pair`: AWS key pair name to attach to the instance
+* `mx_password`: MX password
+* `secure_password`: The password used for communication between the Management Server and the Agent Gateway
+* `dam_version`: Version must be in the format dd.dd.dd.dd where each dd is a number between 1-99 (e.g 14.10.1.10)
+* `management_server_host_for_registration`: DSF DAM Management server address for registration
+
+Refer to [variables.tf](variables.tf) for additional variables with default values and additional info
+
+## Outputs
+
+The following [outputs](outputs.tf) are exported:
+
+* `public_ip`: public address
+* `private_ip`: private address
+* `public_dns`: public dns
+* `private_dns`: private dns
+* `display_name`: Display name of the instance under DSF portal
```

Comment:
I think it's better if we don't use the "DSF portal" term although it is valid and use the "DSF web console" term which we already use.